### PR TITLE
C374173 Verify filtering Requests by Tags with one special symbol (vega) (TaaS)

### PR DESCRIPTION
**Pull request by dzmitryshaplyka1**
[C374173](https://foliotest.testrail.io/index.php?/cases/view/374173)
[Allure report](https://jenkins-aws.indexdata.com/job/folioRancher/job/folioTestingTools/job/runCypressTests/6204/allure/#suites/477c91ba47ec1bbc5ce05a79d4a49d4f/bb3ef93a878e659e/)
Screenshot 2023-12-21 at 13 49 03

### CODE REVIEW

File: cypress/support/fragments/requests/requests.js [hunk 1]
```diff
@@ -314,6 +326,12 @@ export default {
     cy.wait(2000);
   },
 
+  addNewTag(newTag) {
+    cy.do(cy.get('div#input-tag').click().wait(2000).type(newTag));
```
Comment (manowarqqq) [inline cypress/support/fragments/requests/requests.js:330]: why not interactor?

File: cypress/support/fragments/requests/requests.js [hunk 2]
```diff
@@ -299,10 +301,20 @@ export default {
   },
 
   filterRequestsByTag(tag) {
+    cy.wait(2000);
```
Comment (manowarqqq) [inline cypress/support/fragments/requests/requests.js:304]: why do you need waiter here?

File: cypress/support/fragments/requests/requests.js (hunk 1 above)
Comment (dzmitryshaplyka1) [reply cypress/support/fragments/requests/requests.js:330]: @manowarqqq  Frankly speaking, I've tried Interactors and other ways but this component is laggy on Cypress and the only way I find to deal with it in a stable manner is this one.

File: cypress/support/fragments/requests/requests.js (hunk 2 above)
Comment (dzmitryshaplyka1) [reply cypress/support/fragments/requests/requests.js:304]: @manowarqqq same as above, the method is not stable (even on other tests where it is used). Added it to properly deal with Tag chooser.

File: cypress/support/fragments/requests/requests.js [hunk 3]
```diff
@@ -324,12 +335,17 @@ export default {
     cy.wait(2000);
   },
 
+  addNewTag(newTag) {
+    cy.do(cy.get('div#input-tag').click().wait(2000).type(newTag));
```
Comment (re-jas) [inline cypress/support/fragments/requests/requests.js:339]: Please try to use this one
```
cy.do([
      MultiSelect({ id: 'input-tag' }).fillIn(tag),
      MultiSelect({ id: 'input-tag' }).open(),
      MultiSelectOption(including(tag)).click(),
]);
```

**Review by re-jas — APPROVED** [review]

**Comment by dzmitryshaplyka1** [comment]
@manowarqqq  could you please recheck?
thank you!

File: cypress/support/fragments/requests/requests.js [hunk 4]
```diff
@@ -314,10 +315,20 @@ export default {
...
+    cy.wait(2000);
     cy.do(
       Pane({ title: 'Search & filter' })
         .find(MultiSelect({ ariaLabelledby: 'tags' }))
-        .select(tag),
+        .choose(tag),
+    );
+  },
+
+  clearSelectedTag() {
```
Comment (manowarqqq) [inline cypress/support/fragments/requests/requests.js:326]: method will failed if select more than 1 tag
btw in test you need click 'xButton' next to Tags accordion
see resetTypeOfHeading() method as example

File: cypress/e2e/requests/filter-requests-with-special-symbols.cy.js [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+      Requests.closePane('Request Detail');
+      Requests.resetAllFilters();
+      Requests.filterRequestsByTag(tag);
+      Requests.selectFirstRequest(itemData.instanceTitle);
+      Requests.openTagsPane();
+      Requests.verifyAssignedTags(tag);
+
+      Requests.clearSelectedTag();
+      Requests.verifyNoResultMessage('Choose a filter or enter a search query to show results.');
+      Requests.filterRequestsByTag(tag);
```
Comment (manowarqqq) [inline cypress/e2e/requests/filter-requests-with-special-symbols.cy.js:148]: In step 6 and 8 you need to choose  a tag with two different ways
See enterTypeOfHeading and chooseTypeOfHeading as example

File: cypress/support/fragments/requests/requests.js [hunk 6]
```diff
@@ -329,12 +340,20 @@ export default {
     cy.wait(2000);
   },
 
+  addNewTag(tag) {
+    cy.do([
+      addTagInput.fillIn(tag),
+      cy.wait(3000),
+      addTagInput.open(),
```
Comment (manowarqqq) [inline cypress/support/fragments/requests/requests.js:347]: I think addTagInput.open(), is redundant. You already fillIn it

**Review by manowarqqq — APPROVED** [review]

**Comment by dzmitryshaplyka1** [comment]
New [Allure result](https://jenkins-aws.indexdata.com/job/folioRancher/job/folioTestingTools/job/runCypressTests/6376/allure/#suites/477c91ba47ec1bbc5ce05a79d4a49d4f/4edaacad6c72304e/)
Screenshot 2023-12-27 at 16 00 55

**Pull request merged**